src/bun.js/builtins/js/ProcessObjectInternals.js

```diff
@@ -509,20 +510,38 @@ function getStdinStream(fd, rawRequire, Bun) {
       }
     }
 
+    on(name, callback) {
+      // Streams don't generally required to present any data when only
+      // `readable` events are present, i.e. `readableFlowing === false`
+      //
+      // However, Node.js has a this quirk whereby `process.stdin.read()`
+      // blocks under TTY mode, thus looping `.read()` in this particular
+      // case would not result in truncation.
+      //
+      // Therefore the following hack is only specific to `process.stdin`
+      // and does not apply to the underlying Stream implementation.
+      if (name === "readable") {
+        this.ref();
+        this.#unrefOnRead = true;
+      }
+      return super.on(name, callback);
+    }
+
     pause() {
       this.unref();
       return super.pause();
     }
 
     resume() {
-      this.#reader ??= Bun.stdin.stream().getReader();
       this.ref();
       return super.resume();
     }
 
     ref() {
+      this.#reader ??= Bun.stdin.stream().getReader();
```

Comment:
There's supposed to be a way to ref() it directly from the stream. It should auto-unref when stdin itself closes. But, that might lead to a race condition in this case

Reply:
The original code uses `Readable.fromWeb(Bun.stdin.stream())`, and that wrapper class does implement `.[un]ref()`:
https://github.com/oven-sh/bun/blob/2b622162ef6fc999ece19a9bbbc265af326b84a7/src/bun.js/streams.exports.js#L5857-L5858
https://github.com/oven-sh/bun/blob/2b622162ef6fc999ece19a9bbbc265af326b84a7/src/bun.js/streams.exports.js#L6093-L6113

However it doesn't do what I'd expect it to, i.e. preventing the process from exiting:
```js
$ cat test.js
var stdin = Bun.stdin.stream();
var stream = require("stream").Readable.fromWeb(stdin);
console.log(stream.ref, stream.unref);
stream.ref();
stream.on("data", console.log);
// deliberating not calling `.unref()`
// expect process to stick around
```
Typed `abc<Enter>`
```console
$ bun test.js
[Function: ref] [Function: unref]
abc
$
```

Reply:
IMHO if we resolve #1607 all these `.ref()` here or in the underlying `streams` implementations would no longer be necessary, as there should be no additional keep-alive requirements in addition to `bun`/Web API.

Unless of course I misunderstood the purpose of `.ref()`, as I infer it from [Node.js Timers](https://nodejs.org/dist/latest-v18.x/docs/api/timers.html#timeoutref)